internal/templateutil/state.go

```diff
@@ -105,34 +100,25 @@ func ParseAccessRules(ctx context.Context, cl client.Client, rules []hmc.AccessR
 		expectedState.ServiceTemplatesState = make(map[string]map[string]bool)
 	}
 	for _, rule := range rules {
+		supportedClusterTemplates, err := getSupportedTemplates(ctx, cl, rule, hmc.ClusterTemplateKind)
+		if err != nil {
+			errs = errors.Join(errs, err)
+			continue
+		}
+		supportedServiceTemplates, err := getSupportedTemplates(ctx, cl, rule, hmc.ServiceTemplateKind)
+		if err != nil {
+			errs = errors.Join(errs, err)
+			continue
+		}
 		var clusterTemplates []string
 		var serviceTemplates []string
```

Comment:
nit: i suggest preallocating both the slices with the corresponding lengths

Reply:
Oh, no. You're right. I looked at the old code.